tusclient/request.py

```diff
@@ -60,7 +60,7 @@ def perform(self):
             self._response = self.handle.getresponse()
             self.status_code = self._response.status
             self.response_headers = {k.lower(): v for k, v in self._response.getheaders()}
-        except http.client.HTTPException as e:
+        except (http.client.HTTPException, EnvironmentError) as e:
```

Comment:
should we maybe catch this instead? [RemoteDisconnected](https://docs.python.org/3/library/http.client.html#http.client.RemoteDisconnected) I expect that it should be available for both python 2 and 3. I am afraid that `EnvironmentError` might be too broad, and it might be catching something that we don't want.

Reply:
hey @ifedapoolarewaju, thanks for the response.  Understand your concern, but a `RemoteDisconnected` would not have helped me.  In my case, the tusd server drained existing connections and then restarted.  So all existing `http.client` requests worked fine, but I was getting a `ConnectionRefusedError` for the next request upon attempting to upload the next chunk. It was not a subclass of `http.client.HTTPException`.

Reply:
@bgardiner can you please paste the stacktrace error here? I'm thinking there should be a base error common to both `v2` and `v3` thrown when this issue occurs. Or is this error not caught by the `http.client` at all?

Reply:
@ifedapoolarewaju 
>can you please paste the stacktrace error here? 

please see below
> I'm thinking there should be a base error common to both v2 and v3 thrown when this issue occurs. 

Interested to hear what you think that might be

> Or is this error not caught by the http.client at all?

Correct.  It is not caught by http.client at all

---
``` python
...
File "<redacted>", line <redacted>, in <redacted>
  uploader.upload_chunk()
File "/usr/local/lib/python3.6/site-packages/tusclient/uploader.py", line 276, in upload_chunk
  self._do_request()
File "/usr/local/lib/python3.6/site-packages/tusclient/uploader.py", line 287, in _do_request
  self.request.perform()
File "/usr/local/lib/python3.6/site-packages/tusclient/request.py", line 59, in perform
  self.handle.request("PATCH", path, self.file.read(self._content_length), self._request_headers)
File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1239, in request
  self._send_request(method, url, body, headers, encode_chunked)
File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1285, in _send_request
  self.endheaders(body, encode_chunked=encode_chunked)
File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1234, in endheaders
  self._send_output(message_body, encode_chunked=encode_chunked)
File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 1026, in _send_output
  self.send(msg)
File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 964, in send
  self.connect()
File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/http/client.py", line 936, in connect
  (self.host,self.port), self.timeout, self.source_address)
File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/socket.py", line 724, in create_connection
  raise err
File "/usr/local/Cellar/python/3.6.4_3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/socket.py", line 713, in create_connection
  sock.connect(sa)
ConnectionRefusedError: [Errno 61] Connection refused
```

Reply:
@bgardiner we can use something like [this](https://github.com/unixsurfer/haproxyadmin/commit/8baa13a636bdad13b5a93a1ff62e2b62c4024ecb) which seems to be the commonly used pattern from my investigation.

So it'll be something like this:

```python
import errno
...

except OSError as e:
    if e.errno == errno.ECONNREFUSED:
         # retry
    else:
        raise e
```

what do you think?

Reply:
@ifedapoolarewaju I like that approach.  I revised the PR accordingly.  I also, expanded the set of `errno` numbers which are matched to cover a couple of other connection related errors that could be reasonably expected.